scenario_player/tasks/base.py

```diff
@@ -70,9 +70,7 @@ def __call__(self, *args, **kwargs):
         self._start_time = time.monotonic()
         try:
             return self._run(*args, **kwargs)
-        except gevent.GreenletExit:
-            pass
-        except BaseException as ex:
```

Comment:
The thinking was that any exception (including `KeyboardInterrupt`) should terminate a task. That could be fixed by adding a signal handler as we have in Raiden.

Reply:
I went back to `BaseException` but made sure that it's reraised after the task is set to failed.